server/src/engine.rs

```diff
@@ -166,13 +166,41 @@ mod tests {
 
     #[test]
     fn test_get_cargo_pricing() {
```

Comment:
Could you add several more requests to the vector and verify that the prices are in the same order?

Reply:
addressed in the latest commit